### PR TITLE
patch Node.prototype.baseURI

### LINKED ISSUE #40: Set currentScript baseURI
Ideally all nodes within an import should get their `baseURI` updated, but probably this could be limited to only scripts, so that `document.currentScript.baseURI` can be used with native or polyfilled imports, w/o checking if `HTMLImports` is defined.

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #40 
This also provides a polyfill for `baseURI`. In browsers where it's not possible to configure `baseURI` on `Node.prototype`, we configure it on `Element.prototype` - technique tested in safari6 and safari8:
safari8
safari6

Also, delayed the instantiation of `Importer` to when the document is ready.

### CODE REVIEW

File: src/html-imports.js [hunk 1]
```diff
@@ -192,18 +192,16 @@
       // events can be done when all resources are ready.
       this.inflight = 0;
       this.dynamicImportsMO = new MutationObserver(m => this.handleMutations(m));
+      // Observe changes on <head>.
+      this.dynamicImportsMO.observe(document.head, {
```
Comment (sorvell) [inline src/html-imports.js:196]: This change seems unrelated to the issue. Why is it important?

File: src/html-imports.js [hunk 2]
```diff
@@ -694,7 +685,28 @@
     document.addEventListener('load', onLoadingDone, true /* useCapture */ );
     document.addEventListener('error', onLoadingDone, true /* useCapture */ );
   } else {
-    new Importer();
+    // Override baseURI so that imported elements' baseURI can be used seemlessly
+    // on native or polyfilled html-imports.
+    // NOTE: a <link rel=import> will have `link.baseURI === link.href`, as the link
+    // itself is used as the `import` document.
+    const native_baseURI = Object.getOwnPropertyDescriptor(Node.prototype, 'baseURI');
```
Comment (sorvell) [inline src/html-imports.js]: By doing prototypical patching, you're ensuring this won't work on older Chrome/Safari that put accessors on instances. I'm not sure it's worth it. Instead, just putting `baseURI` on import script instances seems like it solves the issue more simply. I suggest trying each approach on Safari 8 to check.

**Review by sorvell — CHANGES_REQUESTED** [review]

File: src/html-imports.js (hunk 1 above)
Comment (valdrinkoshi) [reply src/html-imports.js:196]: oh, it's just to delay the instantiation of Importer when the document is ready.
Now when `Importer` is created, we're sure the document is ready (see https://github.com/webcomponents/html-imports/pull/41/files#diff-0481bace88f50edb13cb81c8c158fe87R713)

File: src/html-imports.js [hunk 3]
```diff
@@ -694,7 +685,28 @@
...
+    // NOTE: in safari9 baseURI is not configurable, so we set it on the Element prototype. 
+    const klass = !native_baseURI || native_baseURI.configurable ? Node : Element;
+    Object.defineProperty(klass.prototype, 'baseURI', {
+      get() {
+        const ownerDoc = /** @type {HTMLLinkElement} */ (isImportLink(this) ? this : importForElement(this));
+        if (ownerDoc) return ownerDoc.href;
+        // Use native baseURI if possible.
+        if (native_baseURI) return native_baseURI.get.call(this);
```
Comment (sorvell) [inline src/html-imports.js]: Does this work on Safari 9? I think the getter may be broken there.

Comment (valdrinkoshi) [reply src/html-imports.js]: True! Great catch, will fix

File: src/html-imports.js (hunk 2 above)
Comment (valdrinkoshi) [reply src/html-imports.js]: Tried in safari 8 and safari 6, in both it's possible to patch `Element.prototype.baseURI` 👌

**Comment by valdrinkoshi** [comment]
@sorvell feedback incorporated, prototypical patching tested on safari6 and safari8 (see updated PR description)

**Review by sorvell — APPROVED** [review]
